index.js

```diff
@@ -86,9 +86,10 @@ export default function normalizeUrl(urlString, options) {
 
 	const hasRelativeProtocol = urlString.startsWith('//');
 	const isRelativeUrl = !hasRelativeProtocol && /^\.*\//.test(urlString);
+	const hasCustomProtocol = /^[a-z.+-]*[.+-][a-z.+-]*:\/\//i.test(urlString);
```

Comment:
```suggestion
	const hasCustomProtocol = /^[a-z][a-z.+-]*?[.+-][a-z.+-]*?:/i.test(urlString);
```

a protocol does not need `//` and it has to start with `a-z`: https://stackoverflow.com/questions/3641722/valid-characters-for-uri-schemes

Reply:
Also add a sensible length limit to prevent ReDoS abuse.